tests/Doctrine/Tests/Common/Collections/Expr/CompositeExpressionTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Doctrine\Tests\Common\Collections\Expr;
+
+use Doctrine\Common\Collections\Expr\CompositeExpression;
+use Doctrine\Common\Collections\Expr\ExpressionVisitor;
+use Doctrine\Common\Collections\Expr\Value;
+use PHPUnit_Framework_TestCase as TestCase;
+
+/**
+ * Class CompositeExpressionTest
+ *
+ * PHP Version 5
+ *
+ * @category  PHP
+ * @package   Doctrine\Tests\Common\Collections\Expr
+ * @author    Tobias Oberrauch <hello@tobiasoberrauch.com>
+ *
+ * @covers \Doctrine\Common\Collections\Expr\CompositeExpression
+ */
+class CompositeExpressionTest extends TestCase
+{
+    /**
+     * @return array
+     */
+    public function invalidDataProvider()
+    {
+        return array(
+            array(
+                'expression' => new Value('value'),
+            ),
+            array(
+                'expression' => 'wrong-type',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider invalidDataProvider
+     *
+     * @param $expression
+     * @return void
+     */
+    public function testExceptions($expression)
+    {
+        $this->setExpectedException('\RuntimeException');
```

Comment:
Can you move the setExpêctedException just before the new compositeExpression.
It's just to make sure that it's the new compositeExpression statement that gets tested.